src/execution/execute.js

```diff
@@ -940,40 +940,53 @@ function completeAbstractValue(
   path: ResponsePath,
   result: mixed
 ): mixed {
-  let runtimeType = returnType.resolveType ?
-    returnType.resolveType(result, exeContext.contextValue, info) :
-    defaultResolveTypeFn(result, exeContext.contextValue, info, returnType);
+  const runtimeTypePromise = returnType.resolveType ?
+    Promise.resolve(returnType.resolveType(
+      result,
+      exeContext.contextValue,
+      info
+    )) :
```

Comment:
It's done. It allows fully synchronous abstract type resolution like before.
PS: I've rebased my PR so this diff is outdated